Building with a bundled libdb changes the names of the exported symbols using
preprocessor magic, so go with it.

--- Linux-PAM-0.77/configure.in	2003-07-07 23:04:14.000000000 -0400
+++ Linux-PAM-0.77/configure.in	2003-07-07 23:04:06.000000000 -0400
@@ -225,6 +225,10 @@
 	AC_CHECK_LIB(db, db_create, HAVE_LIBDB=yes ; AC_DEFINE(HAVE_LIBDB),
 	HAVE_LIBDB=no)
 fi
+if test x$HAVE_LIBDB != xyes ; then
+	AC_CHECK_LIB(db, db_create_pam, HAVE_LIBDB=yes ; AC_DEFINE(HAVE_LIBDB),
+	HAVE_LIBDB=no)
+fi
 AC_SUBST(HAVE_LIBDB)
 AC_CHECK_LIB(fl, yylex, yyterminate, HAVE_LIBFL=yes ; AC_DEFINE(HAVE_LIBFL),
 	HAVE_LIBFL=no)
